File: src/test/java/li/klass/fhem/testutil/MockitoRule.java
package li.klass.fhem.testutil;

import org.junit.rules.MethodRule;
import org.junit.runners.model.FrameworkMethod;
import org.junit.runners.model.Statement;
import org.mockito.MockitoAnnotations;

public class MockitoRule implements MethodRule {
    @Override
    public Statement apply(final Statement base, FrameworkMethod frameworkMethod, final Object test) {
        return new Statement() {
            @Override
            public void evaluate() throws Throwable {
                MockitoAnnotations.initMocks(test);
                base.evaluate();
            }
        };
    }
}

File: app/src/test/java/li/klass/fhem/testutil/MockitoRule.kt
package li.klass.fhem.testutil

import org.junit.rules.MethodRule
import org.junit.runners.model.FrameworkMethod
import org.junit.runners.model.Statement
import org.mockito.MockitoAnnotations

class MockitoRule : MethodRule {
    override fun apply(base: Statement, frameworkMethod: FrameworkMethod, test: Any): Statement {
        return object : Statement() {
            @Throws(Throwable::class)
            override fun evaluate() {
                MockitoAnnotations.initMocks(test)
                base.evaluate()
            }
        }
    }
}
